packages/utils/src/hooks/use-controlled-value.js

```diff
@@ -0,0 +1,26 @@
+import { isNil } from 'lodash';
+import { useState } from 'react';
+
+/**
+ * Simplified and improved implementation of useControlledState.
+ *
+ * @param {object} props
+ * @param {string|number} [props.defaultValue]
+ * @param {string|number} [props.value]
+ * @param {() => void} [props.onChange]
```

Comment:
This should use a type parameter as it'll work for any value...
```suggestion
 * @template T
 * @param {object} props
 * @param {T} [props.defaultValue]
 * @param {T} [props.value]
 * @param {(value: T) => void} [props.onChange]
 * @return {[T, (value: T) => void]}
```